pkg/controller/client/client.go

```diff
@@ -2,55 +2,92 @@ package client
 
 import (
 	"encoding/json"
+	"os"
 
+	"github.com/inspr/inspr/pkg/auth"
 	"github.com/inspr/inspr/pkg/controller"
+	"github.com/inspr/inspr/pkg/ierrors"
 	"github.com/inspr/inspr/pkg/rest/request"
 )
 
 // Client implements communication with the Insprd
 type Client struct {
 	HTTPClient *request.Client
+	config     ControllerConfig
+}
+
+// ControllerConfig stores controller configuration for ease of use and posterior verification.
+type ControllerConfig struct {
+	Auth  request.Authenticator
+	Scope string
+	URL   string
 }
 
 // NewControllerClient return a new Client
-func NewControllerClient(url string, auth request.Authenticator) controller.Interface {
-	client := request.NewClient().BaseURL(url).Encoder(json.Marshal).Decoder(request.JSONDecoderGenerator).Authenticator(auth).Build()
+func NewControllerClient(config ControllerConfig) controller.Interface {
+	client := request.NewClient().BaseURL(config.URL).Encoder(json.Marshal).
+		Decoder(request.JSONDecoderGenerator).Authenticator(config.Auth).Build()
 	return &Client{
 		HTTPClient: client,
+		config:     config,
+	}
+}
+
+// GetInClusterConfigs retrieves controller configs from current dApp deployment.
+func GetInClusterConfigs() (*ControllerConfig, error) {
+	url, urlok := os.LookupEnv("INSPR_INSPRD_ADDRESS")
+	scope, scopeok := os.LookupEnv("INSPR_CONTROLLER_SCOPE")
+	token, tknok := os.LookupEnv("INSPR_CONTROLLER_TOKEN")
+	if !urlok || !scopeok || !tknok {
+		return nil, ierrors.NewError().
+			Message("authentication as controller failed. controllers requires following variables: INSPR_INSPRD_ADDRESS, INSPR_CONTROLLER_SCOPE & INSPR_CONTROLLER_TOKEN").
```

Comment:
Might be a good idea to break this line into two so it's not so wide

Reply:
never mind, figured it out